ast/compile.go

```diff
@@ -130,11 +125,14 @@ func (qc *QueryContext) WithImports(imports []*Import) *QueryContext {
 }
 
 // WithInput sets the input on qc.
+// Warning: This API is deprecated and will be removed in future releases.
 func (qc *QueryContext) WithInput(input Value) *QueryContext {
 	if qc == nil {
 		qc = NewQueryContext()
 	}
-	qc.Input = input
+	// QueryContext no longer does anything with the input and will
+	// ignore any request to associate an input with the context.
+	logrus.Warn("QueryContext.WithInput(..) is deprecated and will be removed in future releases.")
```

Comment:
Question for others.. should we just remove this API?

Reply:
Will do.

Thats a good point on the logging, I've been burned by libraries doing that before ☹️